### PR TITLE
Add support for inwx.de API

### PR DESCRIPTION
**Pull request by speedmann**
```
[root@utility ~/.acme.sh]# ./acme.sh --issue --dns dns_inwx -d www.test.swapoff.de -d test.swapoff.de --test
[Thu Dec  7 11:45:50 CET 2017] Using stage ACME_DIRECTORY: https://acme-staging.api.letsencrypt.org/directory
[Thu Dec  7 11:45:51 CET 2017] Creating domain key
[Thu Dec  7 11:45:51 CET 2017] The domain key is here: /root/.acme.sh/www.test.swapoff.de/www.test.swapoff.de.key
[Thu Dec  7 11:45:51 CET 2017] Multi domain='DNS:test.swapoff.de'
[Thu Dec  7 11:45:51 CET 2017] Getting domain auth token for each domain
[Thu Dec  7 11:45:51 CET 2017] Getting webroot for domain='www.test.swapoff.de'
[Thu Dec  7 11:45:51 CET 2017] Getting new-authz for domain='www.test.swapoff.de'
[Thu Dec  7 11:45:52 CET 2017] The new-authz request is ok.
[Thu Dec  7 11:45:52 CET 2017] Getting webroot for domain='test.swapoff.de'
[Thu Dec  7 11:45:52 CET 2017] Getting new-authz for domain='test.swapoff.de'
[Thu Dec  7 11:45:53 CET 2017] The new-authz request is ok.
[Thu Dec  7 11:45:53 CET 2017] Found domain api file: /root/.acme.sh/dnsapi/dns_inwx.sh
[Thu Dec  7 11:45:54 CET 2017] Adding record
[Thu Dec  7 11:45:55 CET 2017] Found domain api file: /root/.acme.sh/dnsapi/dns_inwx.sh
[Thu Dec  7 11:45:56 CET 2017] Adding record
[Thu Dec  7 11:45:57 CET 2017] Sleep 120 seconds for the txt records to take effect
[Thu Dec  7 11:47:59 CET 2017] Verifying:www.test.swapoff.de
[Thu Dec  7 11:48:02 CET 2017] Success
[Thu Dec  7 11:48:02 CET 2017] Verifying:test.swapoff.de
[Thu Dec  7 11:48:05 CET 2017] Success
[Thu Dec  7 11:48:12 CET 2017] Deleting record
[Thu Dec  7 11:48:14 CET 2017] Deleting record
[Thu Dec  7 11:48:14 CET 2017] Verify finished, start to sign.
./acme.sh: line 1818: warning: command substitution: ignored null byte in input
[Thu Dec  7 11:48:15 CET 2017] Cert success.
<cert>
[Thu Dec  7 11:48:15 CET 2017] Your cert is in  /root/.acme.sh/www.test.swapoff.de/www.test.swapoff.de.cer
[Thu Dec  7 11:48:15 CET 2017] Your cert key is in  /root/.acme.sh/www.test.swapoff.de/www.test.swapoff.de.key
[Thu Dec  7 11:48:16 CET 2017] The intermediate CA cert is in  /root/.acme.sh/www.test.swapoff.de/ca.cer
[Thu Dec  7 11:48:16 CET 2017] And the full chain certs is there:  /root/.acme.sh/www.test.swapoff.de/fullchain.cer
```

### CODE REVIEW

File: dnsapi/dns_inwx.sh [hunk 1]
```diff
@@ -0,0 +1,355 @@
+#!/usr/local/bin/bash
```
Comment (Neilpang) [inline dnsapi/dns_inwx.sh]: do not use "bash",  please use 'sh'

File: dnsapi/dns_inwx.sh [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+  _debug "First detect the root zone"
+  if ! _get_root "$fulldomain"; then
+    _err "invalid domain"
+    return 1
+  fi
+  _debug _sub_domain "$_sub_domain"
+  _debug _domain "$_domain"
+  _debug "Getting txt records"
+
+  printf -v xml_content '<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Neilpang) [inline dnsapi/dns_inwx.sh]: do not use "printf -v",  it's not sh compatible.

**Comment by speedmann** [comment]
sorry for wrong shebang, i was quiet sure i did copy it from cloudflare but must have failed there.
Now the CI should also be happy

File: dnsapi/dns_inwx.sh [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+  </params>
+  </methodCall>' "$_domain" "$_sub_domain"
+  response="$(_post "$xml_content" "$INWX_Api" "" "POST")"
+
+  if ! printf "%s" "$response" | grep "Command completed successfully" >/dev/null; then
+    _err "Error could net get txt records"
+    return 1
+  fi
+
+  if ! printf "%s" "$response" | grep "count" -q; then
```
Comment (Neilpang) [inline dnsapi/dns_inwx.sh]: do not use "grep -q"

File: dnsapi/dns_inwx.sh [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+    _info "Adding record"
+    _inwx_add_record "$_domain" "$_sub_domain" "$txtvalue"
+  else
+    _record_id=$(printf '%s' "$response" | sed -nE 's/.*(<member><name>record){1}(.*)(<name>id<\/name><value><int>)([0-9]+){1}.*/\4/p')
```
Comment (Neilpang) [inline dnsapi/dns_inwx.sh]: do no use "sed -E"

File: dnsapi/dns_inwx.sh [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+  if ! _get_root "$fulldomain"; then
+    _err "invalid domain"
+    return 1
+  fi
+  _debug _sub_domain "$_sub_domain"
+  _debug _domain "$_domain"
+
+  _debug "Getting txt records"
+
+  printf -v xml_content '<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Neilpang) [inline dnsapi/dns_inwx.sh]: "printf -v"

File: dnsapi/dns_inwx.sh [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+  </params>
+  </methodCall>' "$_domain" "$_sub_domain"
+  response="$(_post "$xml_content" "$INWX_Api" "" "POST")"
+
+  if ! printf "%s" "$response" | grep "Command completed successfully" >/dev/null; then
+    _err "Error could not get txt records"
+    return 1
+  fi
+
+  if ! printf "%s" "$response" | grep "count" -q; then
```
Comment (Neilpang) [inline dnsapi/dns_inwx.sh]: "-q"

**Comment by speedmann** [comment]
Thank you for your feedback. I will be fixing the mentioned issues.
I would appreciate if you could add the requirements (limitations of sed, grep etc) to the developer guide for dns apis. I was not aware there are such kind of limitations.

Also feel free to close the PR for now, i can create a new one when i fixed all issues

**Comment by speedmann** [comment]
All mentioned issues should now be solved and i hope in a fashion which fits the project.

Thank you for your patience

**Comment by Neilpang** [comment]
please pull the latest code and resolve the conflicts.
Thanks.

File: dnsapi/dns_inwx.sh [hunk 7]
```diff
@@ -0,0 +1,355 @@
+#!/usr/local/bin/sh
```
Comment (Neilpang) [inline dnsapi/dns_inwx.sh]: it's `#!/usr/bin/env sh`

File: README.md [hunk 8]
```diff
@@ -339,6 +339,7 @@ You don't have to do anything manually!
 1. Dyn Managed DNS API
 1. Yandex PDD API (https://pdd.yandex.ru)
 1. Hurricane Electric DNS service (https://dns.he.net)
+1. INWX (https://www.inwx.de/)
```
Comment (Neilpang) [inline README.md]: please move behind UnoEuro api.  Let's keep in order.
Thanks.

File: dnsapi/README.md [hunk 9]
```diff
@@ -618,6 +618,23 @@ acme.sh --issue --dns dns_unoeuro -d example.com -d www.example.com
 
 The `UNO_Key` and `UNO_User` will be saved in `~/.acme.sh/account.conf` and will be reused when needed.
 
+## 33. Use INWX
+
+INWX offers an xmlrpc api with your standard login credentials, set them like so:
+
```
Comment (Neilpang) [inline dnsapi/README.md]: can you put their website link of INWX here ?

Comment (speedmann) [reply dnsapi/README.md]: To their website or API Documentation?

Comment (Neilpang) [reply dnsapi/README.md]: Maybe both.  just make it more clear to someone like me.
Thanks.

**Pull request merged**